module/zfs/zio.c

```diff
@@ -3984,6 +3985,9 @@ zio_vdev_io_assess(zio_t *zio)
 	 */
 	if (zio->io_error == ENXIO && zio->io_type == ZIO_TYPE_WRITE &&
 	    vd != NULL && !vd->vdev_ops->vdev_op_leaf) {
+		vdev_dbgmsg(vd, "zio_vdev_io_assess(zio=%llx) setting "
```

Comment:
```suggestion
		vdev_dbgmsg(vd, "zio_vdev_io_assess(zio=%px) setting "
```

`%px` is the preferred format specifier here if we really want to log the address.  FreeBSD seems to be fine with it as well since we use it elsewhere.

https://www.kernel.org/doc/html/v4.17/core-api/printk-formats.html

Reply:
I wonder if that's going to work on FreeBSD.  I'll commit your suggestion and we'll see.

Reply:
Since we're already doing it elsewhere I'd assume it should.